tests/test_impls.py

```diff
@@ -50,18 +62,120 @@ def check_initial_contents(filename):
 
 
 @pytest.mark.parametrize(
-    ["initial_req", "cuda_suffix", "cuda_major", "expected_req"],
+    ["cuda_version", "cuda_suffix", "cuda_python_requirement"],
     [
-        ("cupy", "", "11", "cupy-cuda11x"),
-        ("cupy", "", "12", "cupy-cuda12x"),
-        ("cupy", "", None, "cupy"),
-        *((name, "-cu11", "11", f"{name}-cu11") for name in _VERSIONED_RAPIDS_WHEELS),
-        *((name, "-cu12", "12", f"{name}-cu12") for name in _VERSIONED_RAPIDS_WHEELS),
-        *((name, "", None, name) for name in _VERSIONED_RAPIDS_WHEELS),
-        ("poetry", "-cu11", "11", "poetry"),
-        ("poetry", "-cu12", "12", "poetry"),
-        ("poetry", "", None, "poetry"),
+        (("11", "5"), "-cu11", "cuda-python>=11.5,<11.6.dev0"),
+        (("12", "1"), "-cu12", "cuda-python>=12.1,<12.2.dev0"),
     ],
 )
-def tests_add_cuda_suffix(initial_req, cuda_suffix, cuda_major, expected_req):
-    assert _add_cuda_suffix(initial_req, cuda_suffix, cuda_major) == expected_req
+def test_edit_pyproject(cuda_version, cuda_suffix, cuda_python_requirement):
+    with tempfile.TemporaryDirectory() as d:
+        original_contents = """[project]
+name = "test-project"
+dependencies = []
+
+[build-system]
+requires = []
+"""
```

Comment:
You can escape the first newline if you want to make it appear more contiguous. Same goes in a few places below.
```suggestion
        original_contents = """\
[project]
name = "test-project"
dependencies = []

[build-system]
requires = []
"""
```